File: metadata/scripts/ground_type.py
import requests

def obtener_uso_suelo(lat_min, lon_min, lat_max, lon_max):
    """
    Consulta OpenStreetMap vía Overpass API para obtener áreas con 'landuse'
    dentro del bounding box dado.
    Retorna un dict JSON con los elementos.
    """
    overpass_url = "http://overpass-api.de/api/interpreter"
    query = f"""
    [out:json][timeout:25];
    (
      way["landuse"]({lat_min},{lon_min},{lat_max},{lon_max});
      relation["landuse"]({lat_min},{lon_min},{lat_max},{lon_max});
    );
    out body;
    >;
    out skel qt;
    """
    response = requests.get(overpass_url, params={'data': query})
    if response.status_code != 200:
        print("Error en la consulta:", response.status_code, response.text)
        return None
    return response.json()

if __name__ == "__main__":
    # Ejemplo: Región del Biobío (bounding box aproximado)
    lat_min, lon_min = -37.5, -73.5
    lat_max, lon_max = -36.5, -72.5
    uso_suelo = obtener_uso_suelo(lat_min, lon_min, lat_max, lon_max)
    if uso_suelo:
        print("Número de áreas encontradas:", len(uso_suelo.get('elements', [])))
        # Mostrar algunos ejemplos
        for e in uso_suelo.get('elements', [])[:5]:
            print("ID:", e.get('id'), "Tags:", e.get('tags'))

File: metadata/ground_type.js
const axios = require('axios');

async function obtenerUsoSuelo(latMin, lonMin, latMax, lonMax) {
    /**
     * Consulta OpenStreetMap vía Overpass API para obtener áreas con 'landuse'
     * dentro del bounding box dado.
     * Retorna un objeto JSON con los elementos.
     */
    const overpassUrl = "https://overpass-api.de/api/interpreter";
    
    // Query muy simple: solo tags de landuse, sin geometría
    const query = `
    [out:json][timeout:180][bbox:${latMin},${lonMin},${latMax},${lonMax}];
    way["landuse"];
    out tags;
    `;

    try {
        console.log("Consultando Overpass API para uso de suelo...");
        const response = await axios.post(
            overpassUrl,
            `data=${encodeURIComponent(query)}`,
            {
                headers: {
                    'Content-Type': 'application/x-www-form-urlencoded'
                },
                timeout: 200000 // 200 segundos
            }
        );

        if (response.status !== 200) {
            console.error("Error en la consulta:", response.status);
            return null;
        }

        console.log("✓ Datos de uso de suelo obtenidos exitosamente");
        return response.data;
    } catch (error) {
        if (error.code === 'ECONNABORTED' || error.response?.status === 504) {
            console.error("Timeout: El área es demasiado grande. Intenta con un bounding box más pequeño.");
        } else {
            console.error("Error al obtener uso de suelo:", error.message);
        }
        return null;
    }
}

async function main() {
    // Área muy pequeña del centro de Concepción (~2km x 2km)
    const latMin = -36.83, lonMin = -73.06;
    const latMax = -36.81, lonMax = -73.04;
    
    console.log(`Consultando uso de suelo en el área: (${latMin},${lonMin}) a (${latMax},${lonMax})`);
    
    const usoSuelo = await obtenerUsoSuelo(latMin, lonMin, latMax, lonMax);
    
    if (usoSuelo) {
        const elementos = usoSuelo.elements || [];
        console.log("Número de áreas encontradas:", elementos.length);
        
        // Mostrar algunos ejemplos
        elementos.slice(0, 5).forEach(e => {
            console.log("ID:", e.id, "Tags:", e.tags);
        });
    }
}

// Exportar la función
module.exports = { obtenerUsoSuelo };

// Si se ejecuta directamente
if (require.main === module) {
    main().catch(console.error);
}
